libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -580,6 +589,18 @@ void RangeFinder::handle_msg(const mavlink_message_t &msg)
     }
 }
 
+#if HAL_MSP_ENABLED
+void RangeFinder::handle_msp(const msp_rangefinder_sensor_t &pkt)
+{
+    uint8_t i;
+    for (i=0; i<num_instances; i++) {
+        if ((drivers[i] != nullptr) && ((Type)params[i].type.get() != Type::NONE)) {
```

Comment:
I wonder why this doesn't check that the backend type is what we expect rather than != NONE

Reply:
fixed, the mavlink one has the same issue though